### PR TITLE
Added Lazy Loader to skimage.metrics module

### LINKED ISSUE #6964: Consistently use lazy loading for all `skimage.*` submodules
### Description:

With `lazy_loader` successfully being used for `skimage`, `skimage.data` and `skimage.filters` why not use it for every of our public submodules? I see no significant disadvantage here (when using the approach with PYI files) and it is what is proposed in [SPEC 1](https://scientific-python.org/specs/spec-0001/).

Feel free to remove the good first issue label if there are concerns. Otherwise I would suggest to tackle this with separate PRs for each module that copy the examples mentioned above.

### PR DESCRIPTION
**Pull request by anamfatima1304**
Fixes #6964 
This PR adds a lazy loader to skimage.metrics module.

### CODE REVIEW

**Comment by lagru** [comment]
Add @anamfatima1304, I've resolved a merge conflict with `main`. To get to the most recent state for this branch or before adding new local commits to this PR you'll need to update your branch with:
```sh
git switch lazy_loading
git pull --ff-only upstream lazy_loading
```
If that doesn't work, please let me know. :)

File: skimage/metrics/__init__.pyi
Comment (lagru) [inline skimage/metrics/__init__.pyi:1]: This file probably needs some reformatting with black. You should be able to do so with
```sh
pre-commit run --files skimage/metrics/__init__.pyi
```

**Comment by mkcor** [comment]
@lagru more likely `origin` instead of `upstream`, right? I mean, in:
```bash
git switch lazy_loading
git pull --ff-only upstream lazy_loading
```

**Comment by lagru** [comment]
Yes, you are totally right. Good catch! :pray: I fixed it to avoid accidental copies.

**Comment by anamfatima1304** [comment]
Thank You @mkcor and @lagru Your support means a lot. I am thankful for your kind suggestions. I ran the above commands in the terminal. I hope it works.

File: skimage/metrics/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,24 @@
+__all__ = [
```
Comment (mkcor) [inline skimage/metrics/__init__.pyi:1]:
```suggestion
# Explicitly setting `__all__` is necessary for type inference engines
# to know which symbols are exported. See
# https://peps.python.org/pep-0484/#stub-files

__all__ = [
```

**Review by mkcor — APPROVED** [review]
Thank you, @anamfatima1304! I've only suggested adding the comment found at the top of all other `__init__.pyi` files (out of consistency).

**Comment by anamfatima1304** [comment]
Thank you @mkcor. I have made the changes you suggested directly from here (You taught this method on another PR). Thank You for approving. Thank You @lagru for your suggestions, guidance, and resolving merge conflicts.

**Comment by anamfatima1304** [comment]
Hello @lagru and @mkcor A test "Test / macos-cp3.10 (pull_request)" is failing. I tried to search for the cause to fix it but couldn't find it.  Can you please guide me about this? I shall be really grateful to you for your guidance. I am extremely sorry for the trouble that I caused.

**Comment by mkcor** [comment]
@anamfatima1304 no worries! I reran the failing job and it passed this time...

**Comment by anamfatima1304** [comment]
Thank You so much @mkcor

**Comment by lagru** [comment]
Thanks @anamfatima1304!

**Pull request merged**